specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx

```diff
@@ -518,6 +522,15 @@ export function QueryComboBox({
                 }
               />
             )}
+            {hasViewButton && (
```

Comment:
also, handle the case when user has no edit permission, but hasEditButton and hasViewButton is true - in that case your code displays two view buttons